core/lib/spree/preferences/configuration.rb

```diff
@@ -38,18 +38,35 @@ class Configuration
     #     defaults. Set via {#load_defaults}
     attr_reader :loaded_defaults
 
+    attr_reader :load_defaults_called
+
     def initialize
       @loaded_defaults = Spree.solidus_version
+      @load_defaults_called = false
     end
 
     # @param [String] Solidus version from which take defaults when not
     # overriden.
     # @see #load_defaults
     def load_defaults(version)
       @loaded_defaults = version
+      @load_defaults_called = true
       reset
     end
 
+    def check_load_defaults_called(instance_constant_name = nil)
+      return if load_defaults_called || !Spree::Core.solidus_installed?
```

Comment:
Not sure to understand why we need to check `!Spree::Core.solidus_installed?` here, can you explain please?

Reply:
This is to avoid the warning being raised when the user runs the install generator `bin/rails g solidus:install` :slightly_smiling_face: It could lead to thinking that there's some problem when in fact, everything would be ok.

Reply:
Oh it was happening with the previous deprecation system we had for Config as well and it was annoying. It's very hard to understand why this is needed though. I suggest three options:

1. Move it (or wrap it) into a more semantic method like (`is_currently_installing_solidus?`)
2. Move this change to a separate commit well documented
3. Both ¯\_(ツ)_/¯

Reply:
I think you're right. `solidus_installed?` was confusing, as the gem `solidus` is, of course, always installed. But in my view `is_currently_installing_solidus?` is also confusing because it doesn't need to be forcefully the case. I renamed it to `has_install_generator_been_run?`, which I think captures better what we're trying to detect.

About moving it to a different commit, the problem is that we need the check to make the CI green. Otherwise, it will fail because of the warning on the test suite. Even if we add both commits in this PR, I think it's always nice to keep a green history at every step. However, we can update the message for the single commit for this PR (once they're squashed). WDYT?

Reply:
I think the new method name makes sense!

For the different commit, I think that now that the method name is addressed, if we want to keep it in the initial commit, it is not a big deal.